plugin.go

```diff
@@ -810,14 +821,35 @@ func (p *ormPlugin) removeChildAssociations(message *generator.Descriptor) bool
 		// Prep the filter for the child objects of this type
 		keys := p.findAssociationKeys(message, typeNames[typeName], field)
 		childFKeyTypeName := ""
-		p.P(`filterObj`, rawFieldType, ` := `, rawFieldType, `ORM{}`)
+		_, _, fieldTypeNameORM := getTypeNames(typeNames[rawFieldType])
+		p.P(`filterObj`, rawFieldType, ` := `, fieldTypeNameORM, `{}`)
 		for k, v := range keys {
 			for _, childField := range typeNames[rawFieldType].GetField() {
 				if strings.EqualFold(childField.GetName(), k) {
 					childFKeyTypeName, _ = p.GoType(message, childField)
 					break
 				}
 			}
+			if typeNames[rawFieldType].Options != nil {
+				ext, err := proto.GetExtension(typeNames[rawFieldType].Options, gorm.E_Opts)
+				opts, valid := ext.(*gorm.GormMessageOptions)
+				if err == nil && valid {
```

Comment:
I'm not sure what `err` means in this context, but I'm guessing it's not that important?